### DISCUSSION TITLE
Information about directives at application start up

### DISCUSSION THREAD

**Discussion by VitalijF** [Netflix/dgs-framework#1598] [General]
I've custom directive which has been defined in schema and also been applied for some input fields

```directive @length(max: Int!) on INPUT_FIELD_DEFINITION```

After code been generated I've not seen any places where this directive present. So question: is it possible to retrieve information about directive and where it's been applied on start up phase

**Comment by congotej** [comment]
Hi @VitalijF! Can you please provide some more information about what you are trying to achieve?

There is now way of retrieving info about a directive at startup. With the limited info you provided, you can possibly leverage [custom instrumentation](https://www.graphql-java.com/documentation/instrumentation/#custom-instrumentation) classes which give you hooks to inspect and hook into various stages of query processing.

**Reply by VitalijF** [reply]
@congotej 
Hello. 
So, I've created custom directive 
```directive @length(max: Int!) on INPUT_FIELD_DEFINITION```
and applied it 
``` 
input TestInput {
    fieldA: String! @length(max: 30)
    fieldB: String! @length(max: 40)
}
```
The main goal which I try to achieve is to load all these directive values at application start up or at least on fetcher level for applying validation on java side.
